pkg/cloudprovider/aws/fleet/packing/packing.go

```diff
@@ -16,17 +16,21 @@ package packing
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"sort"
 
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
 	"go.uber.org/zap"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 )
 
-type PodPacker struct {
+type podPacker struct {
 	ec2 ec2iface.EC2API
 }
 
-// PodPacker helps pack the pods and calculates efficient placement on the instances.
+// Packer helps pack the pods and calculates efficient placement on the instances.
 type Packer interface {
 	Pack(ctx context.Context, pods []*v1.Pod) ([]*Packings, error)
```

Comment:
Why does this take a `context` arg? (it doesn't seem to be used in the specific implementation - is there some expectation that it might be needed later?)

Reply:
Yes, we will need this later when we start calling EC2 APIs.

Reply:
Probably worth throwing in a TODO there just to make the code reviewing easier ...